server/core/src/main/java/org/infinispan/server/core/LifecycleCallbacks.java

```diff
@@ -60,6 +67,23 @@ public void cacheManagerStarting(GlobalComponentRegistry gcr, GlobalConfiguratio
          BinaryTranscoder transcoder = encoderRegistry.getTranscoder(BinaryTranscoder.class);
          transcoder.overrideMarshaller(jbossMarshaller);
       }
+
+      ComponentRef<ExecutorService> executorServiceRef = basicComponentRegistry.getComponent(
+            KnownComponentNames.NON_BLOCKING_EXECUTOR, ExecutorService.class);
+      ExecutorService oldNonBlockingExecutor = executorServiceRef.wired();
+
+      String nodeName = globalConfiguration.transport().nodeName();
+
+      EventLoopGroup eventLoopGroup = NettyTransport.buildEventLoop(Runtime.getRuntime().availableProcessors(),
+            new DefaultThreadFactory("non-blocking-thread-netty" + (nodeName != null ? "-" + nodeName : "")));
+
+      basicComponentRegistry.replaceComponent(KnownComponentNames.NON_BLOCKING_EXECUTOR, eventLoopGroup, true);
+      basicComponentRegistry.rewire();
+      // TODO: need to ask if this is required or not - I don't think so
+      gcr.rewire();
```

Comment:
It's not required. However, other modules may have created caches already, so you'll need to rewire all the cache components as well with `gcr.rewireNamedRegistries()`.

Reply:
This should be before the cacheManager is created though, so maybe it isn't needed?

Reply:
Yeah, you're right, you can just remove the `gcr` line.

There was certainly a time when `cacheManagerStarting()` callbacks were creating caches, but nowadays modules just register internal caches without creating them. I'm pretty sure trying to create one will actually throw an exception, too.

Reply:
👍